src/modules/i18n/containers/I18nProvider.js

```diff
@@ -84,25 +84,16 @@ export class I18nProvider extends React.Component<PropsType, {
     }
   }
 
-  setLanguage (language: string) {
+  initLanguage (language: string) {
     const targetLanguage = this.getPredeterminedLanguage(language)
 
     const fonts = I18nProvider.getSelectedFonts(targetLanguage)
     this.setState({language: targetLanguage, fonts})
     this.props.setUiDirection(RTL_LANGUAGES.includes(targetLanguage) ? 'rtl' : 'ltr')
-
-    // Set i18next language to apps language
-    this.i18n.changeLanguage(targetLanguage)
   }
 
   componentDidMount () {
-    this.setLanguage(this.props.language)
-  }
-
-  componentDidUpdate (prevProps: PropsType) {
-    if (this.props.language !== prevProps.language) {
-      this.setLanguage(this.props.language)
-    }
+    this.initLanguage(this.props.language)
   }
```

Comment:
Why did you remove the `componentDidUpdate` thing?
We would not set the ui direction in the on language change this way and therefore the ui-direction does not correspond to `cityContent.language` in the store (but to the systemLanguage?).
I think we should have a stricter separation of content-language (which is the one in the store at cityContent.language I guess) and system language...
If you want to only take care about the system-language in the I18nProvider, then please remove the mapStateToProps here and rename all `language` occurences to `systemLanguage`.

Reply:
Also we need to think about what we should do with the fonts, but thats another task...

Reply:
I think we should do this in a separate task. The reason why I renamed setLanguage to initLanguage is that during the runtime of the app the system language will never change.

Also setUiDirection is not doing anything right now and we should probably remove it. But we have to dig deeper into RN to make sure it works as expected: https://facebook.github.io/react-native/blog/2016/08/19/right-to-left-support-for-react-native-apps

Reply:
https://issues.integreat-app.de/browse/NATIVE-191